apps/chat/src/components/Chat/ModelList.tsx

```diff
@@ -70,6 +112,69 @@ const ModelGroup = ({
   }, [entities, recentModelsIds, searchTerm, selectedModelId]);
 
   const description = currentEntity.description;
+  const applicationId = currentEntity.id;
+  const isPublishedEntity = publishedApplicationIds.some(
+    (id) => id === applicationId,
+  );
+
+  const menuItems: DisplayMenuItemProps[] = useMemo(
+    () => [
+      {
+        name: 'Edit',
+        dataQa: 'edit',
+        display: !isPublishedEntity,
+        Icon: IconPencilMinus,
+        onClick: (e: React.MouseEvent) => {
```

Comment:
I think it's better to write these methods inside the parent component and pass the methods directly to that component, rather than passing the set state functions